client/src/store/mutationFolder/ui.ts

```diff
@@ -1,33 +1,67 @@
-import { State, defaultTradeData } from "@port-of-mars/client/store/state";
-import { Role, ResourceAmountData } from "@port-of-mars/shared/types";
+import { State, defaultTradeData } from '@port-of-mars/client/store/state';
+import { Role, ResourceAmountData } from '@port-of-mars/shared/types';
+import {
+  ModalViewType,
+  ModalDataType,
+} from '@port-of-mars/client/types/modals';
 
-function SET_PLAYER_INFO_MODAL_VISIBILITY(state: State, payload: { role: Role; visible: boolean }) {
-  state.ui.modalViews.playerInfoModal = payload;
+// NOTE :: CONTROL MODAL VISIBILITY
+
+function SET_MODAL_VISIBLE(
+  state: State,
+  payload: { type: ModalViewType; data: ModalDataType }
+) {
+  state.userInterface.modalView.type = payload.type;
+  state.userInterface.modalView.data = payload.data;
+  state.userInterface.modalView.visible = true;
 }
 
-function SET_TRADE_REQUEST_MODAL_VISIBILITY(state: State, payload: boolean) {
-  state.ui.modalViews.tradeRequestModal.visible = payload;
-  if (payload == false) {
-    state.ui.tradeData = defaultTradeData();
-  }
+function SET_MODAL_HIDDEN(state: State, payload: any) {
+  state.userInterface.modalView.visible = false;
+  state.userInterface.modalView.type = null;
+  state.userInterface.modalView.data = null;
 }
 
-function SET_CARD_MODAL_VISIBILITY(
-  state: State,
-  payload: { visible: boolean; data: { type: string; info: any } }
-) {
-  state.ui.modalViews.cardModal = payload;
+// NOTE :: CONTROL PROFILE MENU VISIBILITY
+
+function SET_PROFILE_MENU_VISIBILITY(state: State, payload: boolean) {
+  state.userInterface.profileMenuView.visible = payload;
+}
+
+// NOTE :: CONTROL POPUP VISIBILITY
+
+function SET_GROUP_POPUP_VISIBILITY(state: State, payload: boolean) {
```

Comment:
i like that informational components (popups, menus, modals) may be better to consolidate controls (like visibility) for into one general set visibility function. i would suggest considering ways to consolidate similar behaviors of these informational components. 

however, for the sake of time, this task might be better suited for a refactor after the pretest